tests/phpunit/Unit/iCalendar/DateParserTest.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace SRF\Tests\iCalendar;
+
+use SRF\iCalendar\DateParser;
+
+/**
+ * @covers \SRF\iCalendar\DateParser
+ * @group semantic-result-formats
+ *
+ * @license GNU GPL v2+
+ * @since 3.1
+ *
+ * @author mwjames
+ */
+class DateParserTest extends \PHPUnit_Framework_TestCase {
+
+	public function testCanConstruct() {
+
+		$this->assertInstanceOf(
+			DateParser::class,
+			new DateParser()
+		);
+	}
+
+	public function testParseDate_Year() {
+
+		$timeValue = $this->getMockBuilder( '\SMWTimeValue' )
+			->disableOriginalConstructor()
+			->getMock();
```

Comment:
Pretty sure you can use createMock rather than this deprecated form

You can also use SMWTimeValue::class

Reply:
I know but the intent is clearer that constructor is disabled, and we don't count lines so it doesn't really matter.

Reply:
Eh... both the authors of PHPUnit and me beg to differ